recipes/imagemagick/all/conanfile.py

```diff
@@ -64,161 +72,207 @@ class ImageMagicConan(ConanFile):
+                "Windows builds of ImageMagick require MFC which cannot currently be sourced from CCI."
+            )
+
+    def build_requirements(self):
+        if self._settings_build.os == "Windows":
+            self.win_bash = True
+            if not self.conf.get("tools.microsoft.bash:path", default=False, check_type=str):
+                self.tool_requires("msys2/cci.latest")
+
     def source(self):
-        tools.get(
+        get(self,
             **self.conan_data["sources"][self.version]["source"],
-            destination=self._source_subfolder,
+            destination=self.source_folder,
             strip_root=True
         )
 
-        if self._is_msvc:
+        if is_msvc(self):
             visualmagick_version = list(
                 self.conan_data["sources"][self.version]["visualmagick"].keys()
             )[0]
-            tools.get(
+            get(self,
                 **self.conan_data["sources"][self.version]["visualmagick"][
                     visualmagick_version
                 ],
                 destination="VisualMagick",
                 strip_root=True
             )
 
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+        if not cross_building(self):
+            env = VirtualRunEnv(self)
+            env.generate(scope="build")
+        tc = AutotoolsToolchain(self)
+        yes_no = lambda v: "yes" if v else "no"
+        tc.configure_args.extend([
+            "--disable-openmp",
+            "--disable-docs",
+            "--with-perl=no",
+            "--with-x=no",
+            "--with-fontconfig=no",
+            f"--enable-shared={yes_no(self.options.shared)}",
+            f"--enable-static={yes_no(not self.options.shared)}",
+            f"--enable-hdri={yes_no(self.options.hdri)}",
+            f"--with-quantum-depth={self.options.quantum_depth}",
+            f"--with-zlib={yes_no(self.options.with_zlib)}",
+            f"--with-bzlib={yes_no(self.options.with_bzlib)}",
+            f"--with-lzma={yes_no(self.options.with_lzma)}",
+            f"--with-lcms={yes_no(self.options.with_lcms)}",
+            f"--with-openexr={yes_no(self.options.with_openexr)}",
+            f"--with-heic={yes_no(self.options.with_heic)}",
+            f"--with-jbig={yes_no(self.options.with_jbig)}",
+            f"--with-jpeg={yes_no(self.options.with_jpeg)}",
+            f"--with-openjp2={yes_no(self.options.with_openjp2)}",
+            f"--with-pango={yes_no(self.options.with_pango)}",
+            f"--with-png={yes_no(self.options.with_png)}",
+            f"--with-tiff={yes_no(self.options.with_tiff)}",
+            f"--with-webp={yes_no(self.options.with_webp)}",
+            f"--with-xml={yes_no(self.options.with_xml2)}",
+            f"--with-freetype={yes_no(self.options.with_freetype)}",
+            f"--with-djvu={yes_no(self.options.with_djvu)}",
+            f"--with-utilities={yes_no(self.options.utilities)}",
```

Comment:
ImageMagick has a lot more optional dependencies. I suggest we list them all, and forcefully disable them. This caused problems on my machine which had some of these dependencies already installed on my system, and imagemagick linked against them which caused broken builds.